File: Interview-Prep/string/Character-filtering/242-valid-anagram/242.py
class Solution(object):
    def isAnagram(self, s, t):
        count = {}
        for c in s:
            if c not in count:
                count[c]=0
            count[c]+=1
        for c in t:
            if c not in count:
                return False
            count[c]-=1
            if count[c]==0:
                count.pop(c)
        return not len(count)
def main():
    sol = Solution()
    print(sol.isAnagram("an", "nan"))
if __name__=="__main__":
    main()

File: Interview-Prep/string/Character-filtering/242-valid-anagram/242.cpp
#include<iostream>
#include<vector>
#include<unordered_map>
using namespace std;
class Solution{
    public:
    bool isAnagram(string s, string t){
        unordered_map<char, int> cnt;
        for(char c: s){
            cnt[c]++;
        }
        for(char c:t){
            if(!cnt.count(c)) return false;
            cnt[c]--;
            if(cnt[c]==0) cnt.erase(c);
        }
        return cnt.size()==0;
    }
};
int main(){
    Solution solution;
    string s= "anagram", t = "nagarama";
    cout<<solution.isAnagram(s,t);
}